os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -563,6 +564,25 @@ void binary_manager_release_binary_sem(int bin_idx)
 		}
 		sem = sq_next(sem);
 	}
+
+	/* Release all holders for user sempahores */
+	sem = (sem_t *)sq_peek(&BIN_SEMLIST(bin_idx));
```

Comment:
We should consider the shared library case also. If sem is in shared library, we should check we need context switching or not.